rskj-core/src/main/java/co/rsk/mine/ForkDetectionDataCalculator.java

```diff
@@ -0,0 +1,117 @@
+
+        return calculateWithBlockHeaders(mainchainBlockHeaders);
+    }
+
+    public byte[] calculateWithBlockHeaders(List<BlockHeader> mainchainBlockHeaders) {
+        if (mainchainBlockHeaders.size() < MIN_MAINCHAIN_SIZE) {
+            return new byte[0];
+        }
+
+        byte[] forkDetectionData = new byte[12];
+
+        byte[] commitToParentsVector = buildCommitToParentsVector(mainchainBlockHeaders);
+        System.arraycopy(commitToParentsVector, 0, forkDetectionData, 0, 7);
+
+        short numberOfUncles = getNumberOfUncles(mainchainBlockHeaders);
+        forkDetectionData[7] = ByteBuffer.allocate(2).putShort(numberOfUncles).array()[1];
+
+        byte[] blockBeingMinedHeight = getBlockBeingMinedHeight(mainchainBlockHeaders);
+        System.arraycopy(blockBeingMinedHeight, 0, forkDetectionData, 8, 4);
+
+        return forkDetectionData;
+    }
+
+    private byte[] buildCommitToParentsVector(List<BlockHeader> mainchainBlocks) {
+        NetworkParameters params = RegTestParams.get();
+        new Context(params);
+
+        long bestBlockHeight = mainchainBlocks.get(0).getNumber();
+        long blockBeingMinedHeight = bestBlockHeight + 1;
+        long cpvStartHeight = (isMultipleOf64(blockBeingMinedHeight) ?
+                                (long)Math.floor((blockBeingMinedHeight - 1) / (double)CPV_JUMP_FACTOR) :
+                                (long)Math.floor(blockBeingMinedHeight / (double)CPV_JUMP_FACTOR))
+                                * CPV_JUMP_FACTOR;
+
+        byte[] commitToParentsVector = new byte[7];
+        for(int i = 0; i < CPV_SIZE; i++){
+            long currentCpvElement = bestBlockHeight - cpvStartHeight + i * CPV_JUMP_FACTOR;
+            BlockHeader blockHeader = mainchainBlocks.get((int)currentCpvElement);
+            byte[] bitcoinBlock = blockHeader.getBitcoinMergedMiningHeader();
+
+            byte[] bitcoinBlockHash = params.getDefaultSerializer().makeBlock(bitcoinBlock).getHash().getBytes();
+            byte leastSignificantByte = bitcoinBlockHash[bitcoinBlockHash.length - 1];
+
+            commitToParentsVector[i] = leastSignificantByte;
+        }
+
+        return commitToParentsVector;
+    }
+
+    private boolean isMultipleOf64(long number){
+        return number % 64 == 0;
+    }
+
+    private short getNumberOfUncles(List<BlockHeader> mainchainBlocks) {
+        // int to short is a safe cast since number of uncles is max 7 and blocks evaluated are at most 32.
+        // Hence, 7 * 32 = 224 and 224 < 255 (max number that fits on a short type variable)
+        return (short)IntStream
+                .range(0, NUMBER_OF_UNCLES)
+                .map(i -> mainchainBlocks.get(i).getUncleCount()).sum();
+    }
+
+    private byte[] getBlockBeingMinedHeight(List<BlockHeader> mainchainBlocks) {
+        long blockBeingMinedHeight = mainchainBlocks.get(0).getNumber() + 1;
+        return ByteBuffer.allocate(4).putInt((int)blockBeingMinedHeight).array();
```

Comment:
The RSKIP doesn't state if this value should be stored in little or big endian. How are RLP values stored ? Are we using the same standard? In any case, if you are going to submit an improvement of the RSKIP110, then specifying the endianess used would be great.

Reply:
Also note that Bitcoin header values endianess may differ from RLP's

Reply:
Just so you know:

> The initial order of a byte buffer is always `BIG_ENDIAN`.

https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html

Reply:
RLP values are stored in Big Endian.
By default, as @collheesi says, Java byte buffer data structure uses Big Endian.
Hence, there are no code changes to do since it's already big endian and RLP compatible.
I would specify the endianness in RSKIP110 after this PR revision finishes.